sw/device/silicon_creator/mask_rom/BUILD

```diff
@@ -274,3 +279,23 @@ opentitan_functest(
         ],
     ),
 )
+
+# TODO: Add verilator test
+opentitan_functest(
+    name = "e2e_bootup_no_rom_ext_signature",
+    srcs = [
+        "mask_rom_test.c",
+    ],
+    cw310 = cw310_params(
+        args = [
+            "--exec=\"console -q -t0\"",
+            "--exec=\"bootstrap $(location {test_bin})\"",
+            "console",
+            "--timeout=3600",
+            "--exit-success='{}'".format(BOOT_FAILURE_MSG),
+            "--exit-failure='PASS!'",
+        ],
```

Comment:
This should probably be `args` to `opentitan_functest` rather than `cw310_params`.  I don't think these params are specific to the CW310 - they should work regardless of the target test aparatus (verilator, cw310, the future hyperdebug board).

Reply:
yeah actually for sim platforms, shouldn't we prefer a backdoor flash load for execution efficiency? Unless, if the test was testing the bootstrap implementation itself, we probably would have a custom host-side test program (rather than use opentitantool itself) since we would presumably be performing only pieces of the bootstrap procedure, to verify specific components. In this case we could share the same host-side test runner and its args across all platforms, but this will require the opentitan_functest be updated to support a different "test runner" (as we discussed offline).

Reply:
We'll want to bootstrap on all supported non-simulation platforms and backdoor-load for simulation platforms.

Perhaps we should consider the rule's behavior wrt to it's native `args` vs. the more specific `{device}_params.args` (e.g. rather than concatenation, prefer more specific args over general args).

We don't have to solve this issue in this issue: it's ok to leave this as-is and create a follow-up task to find the right solution.

Reply:
Just to make sure I have this straight, @cfrantz are you suggesting:

We try to 1) minimize the generic `args` we pass to each `opentitan_functest` invocation (that ultimately get concatenated onto the default `{device}_params.args` list (in the case that **_no_** `{device}_params.args` are provided), and 2) minimize the times we manually set the `{device}_params` for an `opentitan_functest` invocation (in the case where the defaults cannot simply be appended too)? 

If so, then the we maybe should add a more specific arg to the `opentitan_functest` macro that is the "exit-success" regex? (Then the [`_format_list()`](https://github.com/lowRISC/opentitan/blob/738dc12d1c266f8d31416a09672c8d4e93cd0241/rules/opentitan.bzl#L707) function can be used to propagate this to the actual args that get passed to opentitantool.

Or did I interpret that wrong?

Reply:
Currently, the args for each individual test invocation are prepared by concatenating `opentitan_functest.args` together with the individual `target_params.args` for each `target`.

However, we might instead consider whether the preparation of the per-target args should be either `opentitan_functest.args` _or_ `target_params.args` rather than the concatenation thereof.  This would allow one to write an instance of an `opentitan_functest` rule as:

```
opentitan_functest(
    name = "blargg_test",
    srcs = ["blargg_test.c"],
    args = [
        "--exec=\"console -q -t0\"",
        "--exec=\"bootstrap $(location {test_bin})\"",
        "console",
        "--timeout=3600",
        "--exit-success='{}'".format(BOOT_FAILURE_MSG),
        "--exit-failure='PASS!'",
    ],
    verilator = verilator_params(
        args = [
            "console",
            "--timeout=3600",
            "--exit-success='{}'".format(BOOT_FAILURE_MSG),
            "--exit-failure='PASS!'",
        ]       
   ),
)
```

The verilator invocation would use the more limited set of args provided by verilator_params. All other invocations (cw310, hyperdebug, super-mega-ultra-hyper-debug, etc) would use the "default" args.

As of now, the various `target_params` functions supply the correct defaults for their target, so this is moot for most tests.  However, as we grow the set of allowed targets _and_ the set of tests with special success/failure cases, we'll see a bunch of duplicated `args` stanzas per `target`.

Reply:
Ahhhh, got it, I like that idea. I was actually thinking about this a while back too. I think the way you just described would make things more intuitive as well. The use of [`_format_list()`](https://github.com/lowRISC/opentitan/blob/738dc12d1c266f8d31416a09672c8d4e93cd0241/rules/opentitan.bzl#L707) to prepare the args to send to opentitantool was rather confusing when I first looked at it. I think the way just described makes arg sets explicit. Moreover we could also provide a default for the `opentitan_functest.args` to satisfy the common case.

Reply:
Thanks for clarifying!